src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -133,10 +133,12 @@ void PreReleaseRequest<I>::handle_block_writes(int r) {
                << dendl;
   } else if (r < 0) {
     lderr(cct) << "failed to block writes: " << cpp_strerror(r) << dendl;
-    m_image_ctx.io_work_queue->unblock_writes();
-    save_result(r);
-    finish();
-    return;
+    if(r != -EDQUOT) {
```

Comment:
Copy the pattern of `EBLACKLIST` above and instead add a new `else if (r == -EDQUOT || r == -ENOSPC)` clause to log the error and ignore. No need to save the failure since this is a lock release.

Reply:
done, check this plesae